File: 프로그래머스/lv0/120875. 평행/평행.java
class Solution {
    public int solution(int[][] dots) {
        int answer = 0;
        double[][] arr = new double[3][2];
        arr[0][0] = (dots[0][1]-dots[1][1])/(dots[0][0]-dots[1][0]*1.0);
        arr[0][1] = (dots[2][1]-dots[3][1])/(dots[2][0]-dots[3][0]*1.0);
        arr[1][0] = (dots[0][1]-dots[2][1])/(dots[0][0]-dots[2][0]*1.0);
        arr[1][1] = (dots[1][1]-dots[3][1])/(dots[1][0]-dots[3][0]*1.0);
        arr[2][0] = (dots[0][1]-dots[3][1])/(dots[0][0]-dots[3][0]*1.0);
        arr[2][1] = (dots[1][1]-dots[2][1])/(dots[1][0]-dots[2][0]*1.0);
        for(int i=0; i<3; i++){
            if(arr[i][0] == arr[i][1]){
                return answer = 1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120875. 평행/평행.js
function solution(dots) {
    function line(a, b){
        return (dots[a][1]-dots[b][1])/(dots[a][0] - dots[b][0])
    }
    for([a,b,c,d] of [[0,1,2,3],[0,2,1,3],[0,3,1,2]]){
        if(line(a,b)===line(c,d)) return 1;
    }
    return 0;
}

